test/unittests/test_pla_parser.cpp

```diff
@@ -1,80 +1,118 @@
+    EXPECT_EQ(testPla.nOutputs(), 1U);
+    EXPECT_EQ(testPla.ioCube().size(), 1U);
+
+    auto it11 = testPla.ioCube().find(c11);
+
+    EXPECT_TRUE(it11 != testPla.ioCube().end());
+    EXPECT_EQ(it11->second, c1);
 }
 
 TEST_F(PlaParserTest, orTest) {
     std::string circOr = test_circuits_dir + "or.pla";
 
-    EXPECT_EQ(testPla.nInputs(), 0U);
-    EXPECT_EQ(testPla.nOutputs(), 0U);
-    EXPECT_EQ(testPla.ioCube().size(), 0U);
-
     EXPECT_TRUE(read_pla(testPla, circOr));
 
-    EXPECT_EQ(testPla.nInputs(), 3U);
-    EXPECT_EQ(testPla.nOutputs(), 3U);
-    EXPECT_EQ(testPla.ioCube().size(), 4U);
-}
+    EXPECT_EQ(testPla.nInputs(), 2U);
+    EXPECT_EQ(testPla.nOutputs(), 1U);
+    EXPECT_EQ(testPla.ioCube().size(), 2U);
 
-TEST_F(PlaParserTest, exampleTest) {
-    std::string circExample = test_circuits_dir + "example.pla";
+    auto it_1 = testPla.ioCube().find(c_1);
 
-    EXPECT_EQ(testPla.nInputs(), 0U);
-    EXPECT_EQ(testPla.nOutputs(), 0U);
-    EXPECT_EQ(testPla.ioCube().size(), 0U);
+    EXPECT_TRUE(it_1 != testPla.ioCube().end());
+    EXPECT_EQ(it_1->second, c1);
 
-    EXPECT_TRUE(read_pla(testPla, circExample));
+    auto it1_ = testPla.ioCube().find(c1_);
 
-    EXPECT_EQ(testPla.nInputs(), 6U);
-    EXPECT_EQ(testPla.nOutputs(), 6U);
-    EXPECT_EQ(testPla.ioCube().size(), 32U);
+    EXPECT_TRUE(it1_ != testPla.ioCube().end());
+    EXPECT_EQ(it1_->second, c1);
 }
 
 TEST_F(PlaParserTest, example2Test) {
     std::string circExample2 = test_circuits_dir + "example2.pla";
 
-    EXPECT_EQ(testPla.nInputs(), 0U);
-    EXPECT_EQ(testPla.nOutputs(), 0U);
-    EXPECT_EQ(testPla.ioCube().size(), 0U);
-
     EXPECT_TRUE(read_pla(testPla, circExample2));
 
-    EXPECT_EQ(testPla.nInputs(), 4U);
-    EXPECT_EQ(testPla.nOutputs(), 4U);
-    EXPECT_EQ(testPla.ioCube().size(), 8U);
+    EXPECT_EQ(testPla.nInputs(), 3U);
+    EXPECT_EQ(testPla.nOutputs(), 1U);
+    EXPECT_EQ(testPla.ioCube().size(), 2U);
+
+    auto it1_0 = testPla.ioCube().find(c1_0);
+
+    EXPECT_TRUE(it1_0 != testPla.ioCube().end());
+    EXPECT_EQ(it1_0->second, c1);
+
+    auto it101 = testPla.ioCube().find(c101);
+
+    EXPECT_TRUE(it101 != testPla.ioCube().end());
+    EXPECT_EQ(it101->second, c1);
 }
 
 TEST_F(PlaParserTest, example3Test) {
     std::string circExample3 = test_circuits_dir + "example3.pla";
 
-    EXPECT_EQ(testPla.nInputs(), 0U);
-    EXPECT_EQ(testPla.nOutputs(), 0U);
-    EXPECT_EQ(testPla.ioCube().size(), 0U);
-
     EXPECT_TRUE(read_pla(testPla, circExample3));
+
     EXPECT_EQ(testPla.nInputs(), 2U);
-    EXPECT_EQ(testPla.nOutputs(), 2U);
-    EXPECT_EQ(testPla.ioCube().size(), 4U);
+    EXPECT_EQ(testPla.nOutputs(), 1U);
+    EXPECT_EQ(testPla.ioCube().size(), 2U);
+
+    auto it01 = testPla.ioCube().find(c01);
+
+    EXPECT_TRUE(it01 != testPla.ioCube().end());
+    EXPECT_EQ(it01->second, c1);
+
+    auto it10 = testPla.ioCube().find(c10);
+
+    EXPECT_TRUE(it10 != testPla.ioCube().end());
+    EXPECT_EQ(it10->second, c1);
 }
```

Comment:
do these examples add anything more to the test suite? or do the other tests already suffice?
if not, then consider removing them and the respective files.